Search/ObjectToDocumentConverter.php

```diff
@@ -206,6 +206,32 @@ private function populateDocument($document, $object, $fieldMapping, $prefix = '
                 );
             }
 
+            if (\is_array($value) && (isset($value['value']) || isset($value['fields']))) {
+                if (isset($value['value'])) {
+                    $document->addField(
+                        $this->factory->createField(
+                            $prefix . $fieldName,
+                            $value['value'],
+                            $type,
+                            $mapping['stored'],
+                            $mapping['indexed'],
+                            $mapping['aggregate']
+                        )
+                    );
+                }
+
+                if (isset($value['fields'])) {
+                    /** @var Field $field */
+                    foreach ($value['fields'] as $field) {
+                        $field = clone $field;
+                        $field->setName($prefix . $fieldName . '#' . $field->getName());
+                        $document->addField($field);
+                    }
+                }
+
+                continue;
+            }
```

Comment:
@wachterjohannes what do you think of that? This allows converters to return something like below

~~~php
[
    'value' => $id
    'fields' => [
        new Field('id', $id),
        new Field('name', $name),
    ],
]
~~~

Reply:
👍 

add this also to the docs